### PR TITLE
Add descriptive message for TooManyRequestsException

### PR DESCRIPTION
**Pull request by eolivelli**
After upgrading from 4.5 to 4.6 I found a lot of "Unexpected condition" in some benchmarks.

TooManyRequestsException is a new error introduced in 4.6, there is not description, so the default description is "Unexpected condition", without a debugger it is not possible to understand the cause of the error.

### CODE REVIEW

**Comment by eolivelli** [comment]
@sijie I think this should go to 4.6 branch, I got into this after upgrading.
Maybe it would be worth to add a note in the "release notes" of 4.6

//cc @jiazhai

**Comment by sijie** [comment]
+1 for 4.6.1

**Comment by sijie** [comment]
can you add a test case for this, to prevent "we add a new code but forget to add descriptions for it"?

**Comment by eolivelli** [comment]
@sijie sure will do as soon as I have time

**Comment by eolivelli** [comment]
@sijie added a test case, please review I would like to merge it soon, thanks

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @Test
+    public void testGetMessage() throws Exception {
+        Field[] fields = BKException.Code.class.getFields();
+        int count = 0;
+        for (Field f : fields) {
+            if (f.getType() == Integer.TYPE && !f.getName().equals("UNINITIALIZED")) {
+                int code = f.getInt(null);
+                String msg = BKException.getMessage(code);
+                if (code == UnexpectedConditionException) {
+                    assertThat(msg, equalTo("Unexpected condition"));
```
Comment (sijie) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java]: any reason why not use "assertEquals"?

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                } else {
+                    assertThat("failure on code " + f.getName(), msg,
```
Comment (sijie) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java]: any reason why not use `assertNotEquals`?

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                               not(equalTo("Unexpected condition")));
+                }
+                count++;
+            }
+        }
+        // assert that we found at least 1 code other than UnexpectedConditionException
+        assertThat(count, greaterThan(2));
```
Comment (sijie) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java]: any reason not use assertTrue(count > 2)?

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import static org.apache.bookkeeper.client.api.BKException.Code.UnexpectedConditionException;
+import static org.hamcrest.Matchers.equalTo;
```
Comment (sijie) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java]: any reason why not use junit Asserts? does it not meet your requirements?

Comment (eolivelli) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java]: @sijie 
I saw that in another commit we started to use *assertThat*, recently I started to use it even for "equals", usually error messages generated automatically are more readable (like the one for greaterThan)
I have changed this, now using basic assertEquals/assertNotEquals

**Comment by eolivelli** [comment]
@sijie addressed your comments

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java [hunk 5]
```diff
@@ -38,15 +37,14 @@ public void testGetMessage() throws Exception {
                 int code = f.getInt(null);
                 String msg = BKException.getMessage(code);
                 if (code == UnexpectedConditionException) {
-                    assertThat(msg, equalTo("Unexpected condition"));
+                    assertEquals(msg, "Unexpected condition");
```
Comment (sijie) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/client/api/BKExceptionTest.java]: nit: expected message should be the first parameter: assertEquals("Unexpected condition", msg);

**Pull request closed without merging**